core/sql/executor/HBaseClient_JNI.cpp

```diff
@@ -2844,819 +2871,980 @@ HBC_RetCode HBaseClient_JNI::getHbaseTableInfo(const char* tblName,
+    jenv_->PopLocalFrame(NULL);
+    return HBC_ERROR_INSERTROW_PARAM;
   }
-  return rc;
-}
-
-//////////////////////////////////////////////////////////////////////////////
-// 
-//////////////////////////////////////////////////////////////////////////////
-NAString HTableClient_JNI::getLastJavaError()
-{
-  return JavaObjectInterface::getLastJavaError(JavaMethods_[JM_GET_ERROR].methodID);
+  jlong j_htc = (long)htc;
+  jboolean j_useTRex = useTRex;
+  jlong j_tid = transID;  
+  jlong j_ts = timestamp;
+  jboolean j_checkAndPut = checkAndPut;
+  jboolean j_asyncOperation = asyncOperation;
+ 
+  if (hbs)
+    hbs->getTimer().start();
+  tsRecentJMFromJNI = JavaMethods_[JM_HBC_DIRECT_INSERT_ROW].jm_full_name;
+  jboolean jresult = jenv_->CallBooleanMethod(javaObj_, JavaMethods_[JM_HBC_DIRECT_INSERT_ROW].methodID, 
+               	j_htc, js_tblName, j_useTRex, j_tid, jba_rowID, jRow, j_ts, j_checkAndPut, j_asyncOperation);
+  if (hbs) {
+      hbs->incHbaseCalls();
+      if (!asyncOperation)
+         hbs->incMaxHbaseIOTime(hbs->getTimer().stop());
+  }
+  if (jenv_->ExceptionCheck()) {
+    getExceptionDetails();
+    logError(CAT_SQL_HBASE, __FILE__, __LINE__);
+    logError(CAT_SQL_HBASE, "HBaseClient_JNI::insertRow()", getLastError());
+    jenv_->PopLocalFrame(NULL);
+    if (htc != NULL)
+        NADELETE(htc, HTableClient_JNI, heap);
+    return HBC_ERROR_INSERTROW_EXCEPTION;
+  }
+  if (hbs)
+    hbs->incBytesRead(rowID.len + row.len);
+  if (jresult == false) {
+     jenv_->PopLocalFrame(NULL);
+    if (htc != NULL)
+        NADELETE(htc, HTableClient_JNI, heap);
+     return HBC_ERROR_INSERTROW_DUP_ROWID;
+  }
+  *outHtc = htc;
+  jenv_->PopLocalFrame(NULL);
+  return HBC_OK;
 }
 
 //////////////////////////////////////////////////////////////////////////////
 // 
 //////////////////////////////////////////////////////////////////////////////
-HTC_RetCode HTableClient_JNI::startScan(Int64 transID, const Text& startRowID, 
-   const Text& stopRowID, const LIST(HbaseStr) & cols, Int64 timestamp, 
-   bool cacheBlocks, Lng32 numCacheRows, NABoolean preFetch,
-					const LIST(NAString) *inColNamesToFilter, 
-					const LIST(NAString) *inCompareOpList,
-					const LIST(NAString) *inColValuesToCompare,
-					Float32 samplePercent,
-					NABoolean useSnapshotScan,
-					Lng32 snapTimeout,
-					char * snapName ,
-					char * tmpLoc,
-					Lng32 espNum,
-                                        Lng32 versions)
+HBC_RetCode HBaseClient_JNI::insertRows(NAHeap *heap, const char *tableName,
+      ExHbaseAccessStats *hbs, bool useTRex, Int64 transID, short rowIDLen, HbaseStr rowIDs,
+      HbaseStr rows, Int64 timestamp, bool autoFlush, bool asyncOperation,
+      HTableClient_JNI **outHtc)
 {
-  QRLogger::log(CAT_SQL_HBASE, LL_DEBUG, "HTableClient_JNI::startScan() called.");
+  
+  HTableClient_JNI *htc = NULL;
 
+  if (asyncOperation) {
+     HTableClient_JNI *htc = new (heap) HTableClient_JNI(heap, (jobject)-1);
```

Comment:
same comment as before regarding declaration. I am assuming other methods will have this also. Please check.